### PR TITLE
Add new version of snakemake

### PR DESCRIPTION
**Pull request by charmoniumQ**
Closes #29287

### CODE REVIEW

**Review by bernhardkaindl — DISMISSED** [review]
Builds for me, but maybe wait for review from `adamjstewart` a bit (he reviews python packages)

**Review by charmoniumQ — COMMENTED** [review]
Updated comments

File: var/spack/repos/builtin/packages/py-throttler/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-throttler/package.py]: Why is this patch necessary?

Comment (charmoniumQ) [reply var/spack/repos/builtin/packages/py-throttler/package.py]: The `setup.py` references files that are in the Git repository but not the PyPI archived tarball, so the vendored `setup.py` will fail.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-throttler/package.py]: Ah, I was looking at 1.2.2 (released yesterday) which fixes this. Do you want to just add 1.2.2 and drop 1.2.1 and the patch?

File: var/spack/repos/builtin/packages/snakemake/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/snakemake/package.py:19]: setuptools 42+ required now

File: var/spack/repos/builtin/packages/snakemake/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/snakemake/package.py]:
```suggestion
        depends_on("python@3.7:", type=("build", "run"))
```
Or just remove this line since Spack only supports 3.7+

File: var/spack/repos/builtin/packages/snakemake/package.py
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/snakemake/package.py:19]: filelock no longer needed

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/snakemake/package.py:19]: ratelimiter no longer needed

File: var/spack/repos/builtin/packages/snakemake/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/snakemake/package.py]:
```suggestion
        depends_on("py-yte@1", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/snakemake/package.py
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/snakemake/package.py:19]: jinja2@3 required now

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-throttler/package.py
Comment (charmoniumQ) [reply var/spack/repos/builtin/packages/py-throttler/package.py]: Sounds good

File: var/spack/repos/builtin/packages/snakemake/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/snakemake/package.py]: These should probably be `type="run"` or `type=("build", "run")` so that they are available when you load the package.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — DISMISSED** [review]

**Comment by adamjstewart** [comment]
@spackbot fix style

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**
